reactive/kotlinx-coroutines-rx2/src/RxObservable.kt

```diff
@@ -164,4 +184,6 @@ private class RxObservableCoroutine<T: Any>(
     override fun onCancelled(cause: Throwable, handled: Boolean) {
         signalCompleted(cause, handled)
     }
-}
\ No newline at end of file
+}
+
+internal fun Throwable.isFatal() = this is VirtualMachineError || this is ThreadDeath || this is LinkageError
```

Comment:
It's less efficient, but to avoid duplicating this logic with RxJava2's `throwIfFatal`, you could do
```kotlin
internal fun Throwable.isFatal() = try {
  Exceptions.throwIfFatal(this)
  true
} catch (e: Throwable) {
  false
}
```

Since this is only called when handling an exception that has already been determined to not be a `CancellationException`, that overhead is probably not too much of a concern.

Reply:
Seems this is an Rx-specific piece of code in `RxObservable.kt` it seems sensible to go via Rx `throwIfFatal`.